File: 232. Implement Queue using Stacks/MyQueue.java
import java.util.EmptyStackException;
import java.util.Stack;

public class MyQueue {
    private Stack<Integer> stack = new Stack<>();
    
    public MyQueue() {
        
    }
    
    public void push(int x) {
        this.stack.push(x);
    }
    
    public int pop() {
        if (this.empty())
            throw new EmptyStackException();
        Stack<Integer> temp = new Stack<>();
        int value;
        int stackSize = this.stack.size();
        for (int i = 0; i < stackSize - 1; i++){
            value = stack.pop();
            temp.push(value);
        }
        int poped = this.stack.pop();
        int tempStackSize = temp.size();
        for (int i = 0; i < tempStackSize; i++)
            this.stack.push(temp.pop());
        
        return poped;
        
    }
    
    public int peek() {
        if (this.empty())
            throw new EmptyStackException();
        Stack<Integer> temp = new Stack<>();
        int value;
        int stackSize = this.stack.size();
        for (int i = 0; i < stackSize - 1; i++){
            value = stack.pop();
            temp.push(value);
        }
        int poped = this.stack.pop();
        int tempStackSize = temp.size();
        this.stack.push(poped);
        for (int i = 0; i < tempStackSize; i++)
            this.stack.push(temp.pop());
        
        return poped;
        
    }
    
    public boolean empty() {
        boolean result = this.stack.isEmpty();
        return result;
    }

    public static void main(String[] args) {
        MyQueue myQueue = new MyQueue();
        myQueue.push(1);
        myQueue.push(2);
        myQueue.push(3);
        myQueue.push(5);
        System.out.println(myQueue.peek());
        System.out.println(myQueue.pop());
        System.out.println(myQueue.pop());
        
    }

}

File: 232. Implement Queue using Stacks/MyQueue.rs
struct MyQueue {
    stack: Vec<i32>
}


/** 
 * `&self` means the method takes an immutable reference.
 * If you need a mutable reference, change it to `&mut self` instead.
 */
impl MyQueue {
    
    fn rellocate(stack: &mut Vec<i32>, temp: &mut Vec<i32>) {
        while stack.len() > 0 {
            temp.push(stack.pop().unwrap());
        }
    }

    fn new() -> Self {
        MyQueue { stack: Vec::new() }
    }
    
    fn push(&mut self, x: i32) {
        self.stack.push(x);
    }
    
    fn pop(&mut self) -> i32 {
        let mut temp = Vec::new();
        MyQueue::rellocate(&mut self.stack, &mut temp);
        let popped = temp.pop().unwrap();
        MyQueue::rellocate(&mut temp, &mut self.stack);
        return popped;
    }
    
    fn peek(&self) -> i32 {
        self.stack[0]
    }
    
    fn empty(&self) -> bool {
        self.stack.len() == 0
    }
}

/**
 * Your MyQueue object will be instantiated and called as such:
 * let obj = MyQueue::new();
 * obj.push(x);
 * let ret_2: i32 = obj.pop();
 * let ret_3: i32 = obj.peek();
 * let ret_4: bool = obj.empty();
 */